yogstation/code/modules/power/NuclearEngine.dm

```diff
@@ -0,0 +1,453 @@
+	return react.GetFuelAmounts() //So let's pretend we have 2 CRs and 6 FRs, 2 CRs subtracts 2 heat from this tick, but the FRs add 10 more, so you get +10 heat per tick. Heat is good for the reactor, but too much = the big boom boom
+
+//As we know from physics class, efficiency is helped by cooling it down, so at this point in the reaction your efficiency will be helped by some nice n2, for now we'll simulate n2 with vars.
+/obj/machinery/power/NuclearReactor/proc/GetEfficiency(var/temp = 1,var/amt = 1) //As a percent
+	return (1/temp) * amt
+
+/datum/nuclearreaction
+	var/name = "nuclear reaction"
+	var/FuelAmt = 0 //How much fuel
+	var/CombFuelPower = 0 //And combined, how strong are all our fuels
+	var/list/obj/item/twohanded/required/ControlRod/CRS = list()
+	var/list/obj/item/twohanded/required/FuelRod/FRS = list()
+	var/DepletedAmt = 0 //How much of the uranium has been depleted, and thus acts like control rods?
+	var/obj/machinery/power/NuclearReactor/reactor
+
+/datum/nuclearreaction/proc/GetCoolAmounts()
+	var/AmtSubtract
+	var/frum = 0
+	for(var/obj/item/twohanded/required/ControlRod/CR in CRS)
+		if(!istype(CR,/obj/item/twohanded/required/ControlRod))
+			return
+		AmtSubtract += CR.ControlPower
+	for(var/obj/item/twohanded/required/FuelRod/FR in FRS)
+		if(!istype(FR,/obj/item/twohanded/required/FuelRod))
+			return
+		AmtSubtract += FR.FuelPenalty
+		frum ++
+	if(!frum || frum <=0) //No fuel? Drain that heat fast lad
+		AmtSubtract += 2
+	if(AmtSubtract <= 0)
+		AmtSubtract = 0
+	return AmtSubtract
+
+
+/datum/nuclearreaction/proc/GetFuelAmounts()
+	var/AmtAdd
+	for(var/obj/item/twohanded/required/FuelRod/FR in FRS)
+		if(!istype(FR,/obj/item/twohanded/required/FuelRod))
+			return
+		AmtAdd += FR.FuelPower
+	var/final = AmtAdd
+	return final //IE, AmtSubtract with one CR = 1, AmtAdd with 2 Uranium = 4, so resultant reaction power is 3, which also means a heat spike
+
+/datum/nuclearreaction/proc/ConsumeFuel()
+	for(var/obj/item/twohanded/required/FuelRod/FR in FRS)
+		if(FR.loc != reactor || !FR in reactor.contents)
+			FRS -= FR
+		if(!istype(FR,/obj/item/twohanded/required/FuelRod))
+			return
+		if(FR.integrity <= 0) //That rod's spent, make it into DU
+			reactor.say("[FR] died!")
+			DepletedAmt ++
+			qdel(FR)
+			FRS -= FR
+			FR = null
+			var/obj/item/twohanded/required/FuelRod/depleted/spent = new
+			FRS += spent
+			spent.forceMove(reactor)
+			return
+		if(!istype(FR, /obj/item/twohanded/required/FuelRod/depleted)) //Depleted rods don't deplete further, they just interrupt heat
+			FR.integrity -= 10 - reactor.Efficiency/10 // Eg, 100/10 = 10, so integrity -= 0 aka no health lost, 25 = 2.5, so 8.5 health lost, so keep that efficiency up!
+	for(var/obj/item/twohanded/required/ControlRod/CR in CRS)
+		if(CR.loc != reactor || !CR in reactor.contents)
+			CRS -= CR
+		if(!istype(CR,/obj/item/twohanded/required/ControlRod))
+			return
+		if(CR.integrity <= 10) //CR is spent!
+			qdel(CR)
+			reactor.say("[CR] died!")
```

Comment:
qdel AFTER using it. Also put a return here, you're still editing the ``integrity`` of a qdelled object.